core/src/main/java/com/linecorp/armeria/server/ArmeriaHttpHeaders.java

```diff
@@ -0,0 +1,252 @@
+        connectionDisallowedList = toLowercaseMap(valueCharSequenceIterator(HttpHeaderNames.CONNECTION), 8);
+    }
+
+    @Override
+    public String get(String name) {
+        return builder.get(name);
+    }
+
+    @Override
+    public Integer getInt(CharSequence name) {
+        return builder.getInt(name);
+    }
+
+    @Override
+    public int getInt(CharSequence name, int defaultValue) {
+        return builder.getInt(name, defaultValue);
+    }
+
+    @Override
+    public Short getShort(CharSequence name) {
+        return builder.getInt(name).shortValue();
+    }
+
+    @Override
+    public short getShort(CharSequence name, short defaultValue) {
+        return (short) builder.getInt(name, defaultValue);
+    }
+
+    @Override
+    public Long getTimeMillis(CharSequence name) {
+        return builder.getTimeMillis(name);
+    }
+
+    @Override
+    public long getTimeMillis(CharSequence name, long defaultValue) {
+        return builder.getTimeMillis(name, defaultValue);
+    }
+
+    @Override
+    public List<String> getAll(String name) {
+        return builder.getAll(name);
+    }
+
+    @Override
+    public List<Entry<String, String>> entries() {
+        return builder.stream()
+                      .map(e -> new AbstractMap.SimpleEntry<>(e.getKey().toString(), e.getValue()))
+                      .collect(Collectors.toList());
+    }
+
+    @Override
+    public boolean contains(String name) {
+        return builder.contains(name);
+    }
+
+    @Override
+    public Iterator<Entry<String, String>> iterator() {
+        return builder.stream()
+                      .map(e -> (Entry<String, String>) new AbstractMap.SimpleEntry<>(
+                              e.getKey().toString(), e.getValue())
+                      )
+                      .iterator();
+    }
+
+    @Override
+    public Iterator<Entry<CharSequence, CharSequence>> iteratorCharSequence() {
+        return  builder.stream()
+                       .map(e -> (Entry<CharSequence, CharSequence>) new AbstractMap.SimpleEntry<>(
+                               (CharSequence) e.getKey().toString(),(CharSequence) e.getValue())
+                       )
+                       .iterator();
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return builder.isEmpty();
+    }
+
+    @Override
+    public int size() {
+        return builder.size();
+    }
+
+    @Override
+    public Set<String> names() {
+        return builder.names().stream().map(AsciiString::toString).collect(Collectors.toSet());
+    }
+
+    @Override
+    public HttpHeaders add(String name, Object value) {
+        final AsciiString asciiName = AsciiString.of(name);
+
+        if (asciiName.equals(HttpHeaderNames.CONNECTION)) {
+            final AsciiString lowerCased = AsciiString.of(value.toString().toLowerCase());
+            splitByCommaAndAdd(connectionDisallowedList, lowerCased);
+        }
+
+        if (HTTP_TO_HTTP2_HEADER_DISALLOWED_LIST.contains(asciiName) ||
```

Comment:
With this line of code, this headers won't contain `Transfer-Encoding` header.
As a result, below test failed because it considers corresponding request as if `endOfStream`.
<img width="1277" alt="스크린샷 2023-06-08 오후 8 51 41" src="https://github.com/line/armeria/assets/16456651/9d6b7627-b8f6-46de-8844-6bb1a3b3e683">

https://github.com/line/armeria/blob/b808c3e7fb7deb604e46c2c82c189a30ed7822f4/core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java#L262

So I'm trying to solve these kind of regression without copying another header.

Reply:
Oops, I missed this comment. If so, I think we can remove the `Transfer-Encoding` header in `Http1RequestDecoder` after `isTransferEncodingChunked` is executed. WDYT?

Reply:
@minwoox 
I tried another approach that moved `isTransferEncodingChunked` up before filtering out disallowed headers.
https://github.com/line/armeria/pull/4863/commits/a2a2f4853f1bb4bfc48b8ff2d1ddcf6ec7b3cd5e#diff-a3306f0289f199b1657bb0e32849fd97502c82a91318adc373f8e750a96f84ecR187-R188